### PR TITLE
Add PayPal instructions to README

### PR DESCRIPTION
**Pull request by intelliot**
- Steps for setting up PayPal with PopupBridge iOS
- Create `popupbridge-paypaldatacollector-android.md`
  - Code snippet for data collector integration

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -171,6 +165,23 @@ PayPal used to support authentication via a modal iframe, but authentication now
 
 PopupBridge solves this problem by allowing [`braintree-web`](https://github.com/braintree/braintree-web) to open the PayPal popup from a secure mini-browser.
 
+Using PayPal in a WebView
+-------------------------
+
+WebView-based checkout flows can accept PayPal with PopupBridge the Braintree JS SDK. For the authentication flow, PayPal requires a popup window—which can be simulated with PopupBridge.
```
Comment (lkorth) [inline README.md]: "PopupBridge the Braintree JS SDK" => "PopupBridge and the [Braintree JS SDK](link)"

File: README.md [hunk 2]
```diff
@@ -171,6 +165,23 @@ PayPal used to support authentication via a modal iframe, but authentication now
...
+
+### Setup
+1. Create a web-based checkout that accepts PayPal using Braintree JS v3.9.0 or higher
+2. Show a loading indicator
+    - PopupBridge lacks the PayPal loading page that customers see when using PayPal through a real popup window. We recommend adding a loading indicator on your web page when the PayPal button is clicked. In the PayPal tokenize completion callback, hide the loading indicator.
+3. Create a native mobile app that opens the checkout in a WebView
+4. Integrate the PopupBridge library
+5. Collect device data
+    - To help detect fraudulent activity, collect device data before performing PayPal transactions. This is similar to collecting device data with our [native Android SDK](https://developers.braintreepayments.com/guides/paypal/vault/android/v2#collecting-device-data) with a few differences:
+        1. Rather than the entire data collector, you can add just PayPalDataCollector to your app, e.g. `compile 'com.paypal.android.sdk:data-collector:2.4.2'`
```
Comment (lkorth) [inline README.md]: The docs are wrong on https://developers.braintreepayments.com/guides/paypal/vault/android/v2#collecting-device-data and should not include `com.braintreepayments.api:data-collector`. Reword this to Include `PayPalDataCollector` in your `build.gradle` dependencies.

Comment (lkorth) [inline README.md]: 2.+ would be a good for the version here too.

Comment (lkorth) [inline README.md]: This line could actually be removed once the docs are updated.

File: popupbridge-paypaldatacollector-android.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        mWebView = (WebView) findViewById(R.id.web_view);
+
+        mUuid = getOrGenerateUuid();
+        mPopupBridge = PopupBridge.newInstance(this, mWebView);
+
+        mPopupBridge.setNavigationListener(this);
+        mWebView.loadUrl("http://10.0.2.2:3000");
```
Comment (lkorth) [inline popupbridge-paypaldatacollector-android.md]: Drop the port in the url.

File: popupbridge-paypaldatacollector-android.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import com.braintreepayments.popupbridge.PopupBridgeNavigationListener;
+import com.paypal.android.sdk.data.collector.SdkRiskComponent;
+
+import java.util.UUID;
+
+public class MainActivity extends Activity implements PopupBridgeNavigationListener {
+
+    private WebView mWebView;
+    private PopupBridge mPopupBridge;
+    private String mUuid;
```
Comment (lkorth) [inline popupbridge-paypaldatacollector-android.md]: Delete.

File: popupbridge-paypaldatacollector-android.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        mWebView = (WebView) findViewById(R.id.web_view);
+
+        mUuid = getOrGenerateUuid();
```
Comment (lkorth) [inline popupbridge-paypaldatacollector-android.md]: Delete.

File: popupbridge-paypaldatacollector-android.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        mPopupBridge = PopupBridge.newInstance(this, mWebView);
+
+        mPopupBridge.setNavigationListener(this);
+        mWebView.loadUrl("http://10.0.2.2:3000");
+    }
+
+    private String getOrGenerateUuid() {
```
Comment (lkorth) [inline popupbridge-paypaldatacollector-android.md]: Delete this method.

File: popupbridge-paypaldatacollector-android.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            String checkoutId = uri.getQueryParameter("token");
+            String vaultId = uri.getQueryParameter("ba_token");
+
+            if (!TextUtils.isEmpty(checkoutId)) {
+                id = checkoutId;
+            } else if (!TextUtils.isEmpty(vaultId)) {
+                id = vaultId;
+            }
+
+            Log.d("result", SdkRiskComponent.getClientMetadataId(this, mUuid, id));
```
Comment (lkorth) [inline popupbridge-paypaldatacollector-android.md]: This should now be using [`PayPalDataCollector#getClientMetadataId(context, pairingId)`](https://github.com/braintree/braintree_android/blob/7f6171a055218d213dda6cfd544711046780797e/PayPalDataCollector/src/main/java/com/paypal/android/sdk/data/collector/PayPalDataCollector.java#L47). That eliminates the need to generate a UUID.

File: popupbridge-paypaldatacollector-android.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+                    .putString("uuid", savedUuid)
+                    .apply();
+        }
+
+        return savedUuid;
+    }
+
+    @Override
+    public void onUrlOpened(String url) {
+        if (url.startsWith("https://checkout.paypal.com")) {
```
Comment (lkorth) [inline popupbridge-paypaldatacollector-android.md]: `checkout.paypal.com` is only used for future payments. This will fail for single payments and billing agreements.

**Comment by intelliot** [comment]
@lkorth Could you have another look at this? Thanks!

File: README.md [hunk 9]
```diff
@@ -167,10 +161,25 @@ We are a team of engineers who work on the Developer Experience team at [Braintr
 
 Short answer: to accept PayPal as a payment option when mobile apps are using a WebView to power the checkout process.
 
-PayPal used to support authentication via a modal iframe, but authentication now occurs in a popup window to increase user confidence that their account information is protected from malicious actors (the address bar shows `https://checkout.paypal.com` with the HTTPS lock icon). However, this causes issues with Braintree merchants who use a web page to power payments within their apps: they can't accept PayPal because WebViews cannot open popups and return the PayPal payment authorization data to the parent checkout page.
+PayPal used to support authentication via a modal iframe, but authentication now occurs in a popup window to increase user confidence that their account information is protected from malicious actors (the address bar shows `paypal.com` with the HTTPS lock icon). However, this causes issues with Braintree merchants who use a web page to power payments within their apps: they can't accept PayPal because WebViews cannot open popups and return the PayPal payment authorization data to the parent checkout page.
```
Comment (lkorth) [inline README.md]: Mixing can't and cannot is the same sentence sounds weird. Might be worth updating this to one or the other.

File: README.md [hunk 10]
```diff
@@ -167,10 +161,25 @@ We are a team of engineers who work on the Developer Experience team at [Braintr
...
+
+### Setup
+1. Create a web-based checkout that accepts PayPal using Braintree JS v3.9.0 or higher
+2. Show a loading indicator
+    - PopupBridge lacks the PayPal loading page that customers see when using PayPal through a real popup window. We recommend adding a loading indicator on your web page when the PayPal button is clicked. In the PayPal tokenize completion callback, hide the loading indicator.
+3. Create a native mobile app that opens the checkout in a WebView
+4. Integrate the PopupBridge library
+5. Collect device data
+    - To help detect fraudulent activity, collect device data before performing PayPal transactions. This is similar to collecting device data with our [native Android SDK](https://developers.braintreepayments.com/guides/paypal/vault/android/v2#collecting-device-data), and you will need to implement a method in your native app for sending device data. See the [Android code snippet for PayPal + PopupBridge](popupbridge-paypaldatacollector-android.md)
```
Comment (lkorth) [inline README.md]: Maybe get rid of the `and` here? "This is similar to collecting device data with our native Android SDK, you will need to implement a method in your native app for sending device data."

File: README.md [hunk 11]
```diff
@@ -167,10 +161,27 @@ We are a team of engineers who work on the Developer Experience team at [Braintr
...
 PopupBridge solves this problem by allowing [`braintree-web`](https://github.com/braintree/braintree-web) to open the PayPal popup from a secure mini-browser.
 
+Using PayPal in a WebView
+-------------------------
+
+WebView-based checkout flows can accept PayPal with PopupBridge and the [Braintree JS SDK](https://github.com/braintree/braintree-web). For the authentication flow, PayPal requires a popup window—which can be simulated with PopupBridge.
+
+### Setup
+1. Create a web-based checkout that accepts PayPal using Braintree JS v3.9.0 or higher
+2. Show a loading indicator
```
Comment (quinnjn) [inline README.md]: Nitpick but all of these can be `1.`. Keeps the numbered list but avoids having to update all the numbers.

1. This
1. Is
1. A
     1. Nested
     1. List
1. Example
